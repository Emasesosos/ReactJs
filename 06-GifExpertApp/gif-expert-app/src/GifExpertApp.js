import React, { useState } from 'react';
import AddCategory from './components/AddCategory';
import GifGrid from './components/GifGrid';
import Imagen from './components/Imagen';

const GifExpertApp = ({defaultCategories = []}) => {

    // const [categories, setCategories] = useState(['One Punch']);
    const [categories, setCategories] = useState(defaultCategories);

    return (
        <div className="gif-expert-app">
            <div className="main-title">
                <Imagen />
                <h2>GifExpertApp</h2>
                <Imagen />
            </div>
            <AddCategory 
                setCategories={setCategories}
            />
            <hr/>
            <ol>
                {
                    categories.map((category) => {
                        return <GifGrid
                                    key={category}
                                    category={category}
                                />;
                    })
                }
            </ol>
        </div>
    );
}

export default GifExpertApp;