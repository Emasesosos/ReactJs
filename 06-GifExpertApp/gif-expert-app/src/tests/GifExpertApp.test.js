import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import GifExpertApp from './../GifExpertApp';

describe('Pruebas en <GifExpertApp />', () => {

    test('Debe de mostrar el componente correctamente', () => {
        const wrapper = shallow( <GifExpertApp /> );
        expect(wrapper).toMatchSnapshot();
    });

    test('Debe de mostrar una lista de categorias', () => {
        const categories = ['One Punch', 'Dragon Ball', 'Black Clover'];
        const wrapper = shallow( <GifExpertApp defaultCategories={categories}/> );

        expect(wrapper).toMatchSnapshot();
        const GifGrid = wrapper.find('GifGrid');
        expect(GifGrid.length).toBe(categories.length);
    });

});