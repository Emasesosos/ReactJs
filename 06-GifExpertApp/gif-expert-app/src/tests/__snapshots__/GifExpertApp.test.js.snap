// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifExpertApp /> Debe de mostrar el componente correctamente 1`] = `
<div
  className="gif-expert-app"
>
  <div
    className="main-title"
  >
    <Imagen />
    <h2>
      GifExpertApp
    </h2>
    <Imagen />
  </div>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</div>
`;

exports[`Pruebas en <GifExpertApp /> Debe de mostrar una lista de categorias 1`] = `
<div
  className="gif-expert-app"
>
  <div
    className="main-title"
  >
    <Imagen />
    <h2>
      GifExpertApp
    </h2>
    <Imagen />
  </div>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="One Punch"
      key="One Punch"
    />
    <GifGrid
      category="Dragon Ball"
      key="Dragon Ball"
    />
    <GifGrid
      category="Black Clover"
      key="Black Clover"
    />
  </ol>
</div>
`;
